docs/contributor/arch/018-switch-from-gateways-to-a-central-agent.md

```diff
@@ -0,0 +1,72 @@
+- separation of concerns
+  It is always good to have clean responsibilities to keep code better maintainable. Here, the gateways will do enrichment and dispatching, while agents to data collection only
+- decoupling of signal types
+  Processing the signal types in separate processes keep the processing tolerant to failures. An overload of metrics will keep the log delivery stable.
+- tail-based sampling
+  Scenarios like trace sampling requires processing all spans of a trace at one instance, requiring a StatefulSet with sticky trace routing. However, that scenario might not be relevant as this will be done on the backend usually.
+
+The drawback of Gateways are:
+- no uniform push endpoint
+  by design there is a gateway per signal type were the application need to communicate to. With that you need to configure an individual OTLP endpoint per signal type always. Having an uniform URL will require an additional component route the requests.
+- no localhost support
+  A typical approach is to run an agent per node having a host port available where the node-local applications can push the data to. That communication will be very reliable and fast and requires no encryption. Also, the default setting of the otel-sdk will work at any time, no kyma specific endpoint required at all.
+- autoscaling is tricky
+  The gateway should provide autoscaling capabilities which are tricky to realize. It must be detected very fast if a scale out is needed and it situations were the backend is overloaded must be excluded from scale out.
+- every instance caches whole cluster state
+  Every instance of a gateway must be able to enrich data of every pod and with that need to cache all pod metadata
+- Istio mandatory for load balancing
+  Pushing OTLP data from an application to the gateway running with replicas requires loadbalancing. Unfortunately, a kubernetes service is acting on L4 and while GRPC/HTTP2 is on L/, so an additional loadbalancer is needed. That can be easily solved by using Istio, but it becomes a mandatory requirement for applications. However, there can be applications not using Istio (Statefulets often don't do) and like to provide OTLP data.
+- Istio is mandatory for mTLS
+  App to Gateway communication is across nodes and with that should use mTLS, so Istio needs to be used which sometimes is not possible. Also, Istio causes additional overhead like noise filtering
+- cumulativetodelta transformation not suppoerted
+  This transformation is mandatory for Dynatrace support and requires that data from a pod gets processed always by the same instance
+- General misconception
+  Gateways are usually running outside the cluster as a first entry point. The cluster should get rid of the data as fast as possible as persistence cannot be provided here (thinks like Kafka would again run outside). Try to compensate for short-lived hickups but otherwise just point back the backpressure to the client and do not try to solve the big picture
```

Comment:
```suggestion

  Gateways are usually running outside the cluster as a first entry point. The cluster should get rid of the data as fast as possible as persistence cannot be provided here (things like Kafka would again run outside). Try to compensate for short-lived hiccups. Otherwise, point back the backpressure to the client and do not try to solve the big picture.
```